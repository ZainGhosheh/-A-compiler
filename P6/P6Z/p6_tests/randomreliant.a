main : () -> void {
  a:int;
  maybe a means 8 otherwise 5;

  b:bool;
  b = eh?;
}