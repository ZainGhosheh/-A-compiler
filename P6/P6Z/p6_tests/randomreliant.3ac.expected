[BEGIN GLOBALS]
main
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN main LOCALS]
a (local var of 8 bytes)
b (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            call randBool
            getret [tmp0]
            IFZ [tmp0] GOTO lbl_2
            [a] := 8
            goto lbl_3
lbl_2:      nop
            [a] := 5
lbl_3:      nop
            call randBool
            getret [tmp1]
            [b] := [tmp1]
lbl_1:      leave main